examples/rag/app_rag_demo.py

```diff
@@ -0,0 +1,479 @@
+        .mark_bar()
+        .encode(
+            x=alt.X(f"p({i+1}|q)", scale=alt.Scale(domain=[0, 0.6]), axis=None),
+            y=alt.Y(f"d:{i+1}", axis=None),
+            color=alt.Color("color", scale=None),
+        )
+    )
+    alt_text = alt_bar.mark_text(
+        align="left",
+        baseline="middle",
+        fontSize=16,
+        fontWeight="bold",
+        dx=5,  # Nudges text to right so it doesn't appear on top of the bar
+    ).encode(text=alt.Text(f"p({i+1}|q)", format=",.3f"))
+    chart = (
+        alt.layer(alt_bar, alt_text).configure_axis(grid=False).configure_view(strokeOpacity=0).properties(width=300)
+    )
+    col_a.write(chart)
+
+
+for j, doc in enumerate(model_outputs["documents"][4:]):
+    i = j + 4
+    wiki_url = f"https://en.wikipedia.org/wiki/{doc['title'].replace(' ', '_')}"
+    title_short = doc["title"] if len(doc["title"]) < 26 else doc["title"][:24] + "..."
+    title_url = f"[{title_short}]({wiki_url})"
+    col_b.markdown(
+        f"<font color={colors[i]}>&#11044; -- Article {i+1} --</font> {title_url}",
+        unsafe_allow_html=True,
+    )
+    df_s = pd.DataFrame({f"d:{i+1}": [f"{i+1}"], f"p({i+1}|q)": [doc["proba"]], "color": colors[i]})
+    alt_bar = (
+        alt.Chart(df_s)
+        .mark_bar()
+        .encode(
+            x=alt.X(f"p({i+1}|q)", scale=alt.Scale(domain=[0, 0.6]), axis=None),
+            y=alt.Y(f"d:{i+1}", axis=None),
+            color=alt.Color("color", scale=None),
+        )
+    )
+    alt_text = alt_bar.mark_text(
+        align="left",
+        baseline="middle",
+        fontSize=16,
+        fontWeight="bold",
+        dx=5,  # Nudges text to right so it doesn't appear on top of the bar
+    ).encode(text=alt.Text(f"p({i+1}|q)", format=",.3f"))
+    chart = (
+        alt.layer(alt_bar, alt_text).configure_axis(grid=False).configure_view(strokeOpacity=0).properties(width=300)
+    )
+    col_b.write(chart)
+
+
+# Visualize token-level behavior
+right_c.markdown("--- \n ### Word-level contribution of Wikipedia passages:")
+right_c.markdown(
+    "> When generating the output, the contribution of each passage to the next token depends both on its retrieval score and on the tokens that have been generated previously. Select a generated output to show token-level provenance information:"
+)
+
+answers = [a for a, l in model_outputs["answers"]]
+losses = [l for a, l in model_outputs["answers"]]
+answer_s = col_c.selectbox(
+    "",
+    answers,
+    index=losses.index(max(losses)),
+)
+
+explained = model_outputs["explanations"][answers.index(answer_s)]
+joined_exp = join_explanation(explained, 2)
+
+# word-level provenance
+joined_an = tuple([(w, str(did + 1), colors[did]) for w, did in joined_exp])
+col_c.markdown(
+    "- "
+    + " ".join([f"<font color={colors[did]}>**{w}**-{did+1}-</font>" for w, did in joined_exp if len(w) > 0])
+    + " \n",
+    unsafe_allow_html=True,
+)
+
+# token-level plot
+plot_container = col_d.container()
```

Comment:
As with the columns, the current function for the container is `beta_container()`.